packages/schematics/angular/migrations/update-12/replace-prod-flag.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { Rule } from '@angular-devkit/schematics';
+import { JSONFile } from '../../utility/json-file';
+
+export default function (): Rule {
+  return (tree) => {
+    const file = new JSONFile(tree, 'package.json');
+    const scripts = file.get(['scripts']);
+    if (!scripts || typeof scripts !== 'object') {
+      return;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const updatedScripts = Object.entries(scripts!).map(([key, value]) => [
+      key,
+      typeof value === 'string' ? value.replace(/ --prod/g, ' --configuration production') : value,
```

Comment:
Maybe require a word boundary after `--prod`?
```suggestion
      typeof value === 'string' ? value.replace(/ --prod(?!\w)/g, ' --configuration production') : value,
```

Or `/ --prod(?=\W)/g` would also work. This avoids false positives for e.g. a `--product` flag.

Reply:
Good catch